x-pack/plugins/security_solution/server/lib/telemetry/filters.ts

```diff
@@ -0,0 +1,146 @@
+import {
+	TelemetryEvent
+} from './types';
+
+// For the Allowlist definition.
+
+interface AllowlistFields {
+	[key: string]: boolean | AllowlistFields;
+}
+
+// Allow list process fields within events.  This includes "process" and "Target.process".'
+const allowlistProcessFields: AllowlistFields = {
+	args: true,
+	name: true,
+	executable: true,
+	command_line: true,
+	hash: true,
+	pid: true,
+	pe: {
+		original_file_name: true,
+	},
+	uptime: true,
+	Ext: {
+		architecture: true,
+		code_signature: true,
+		dll: true,
+		malware_signature: true,
+		token: {
+			integrity_level_name: true,
+		},
+	},
+	thread: true,
+	working_directory: true,
+};
+
+// Allow list for event-related fields, which can also be nested under events[]
+const allowlistBaseEventFields: AllowlistFields = {
+	dll: {
+		name: true,
+		path: true,
+		code_signature: true,
+		malware_signature: true,
+		pe: {
+			original_file_name: true,
+		},
+	},
+	event: true,
+	file: {
+		extension: true,
+		name: true,
+		path: true,
+		size: true,
+		created: true,
+		accessed: true,
+		mtime: true,
+		directory: true,
+		hash: true,
+		Ext: {
+			code_signature: true,
+			header_data: true,
+			malware_classification: true,
+			malware_signature: true,
+			quarantine_result: true,
+			quarantine_message: true,
+		},
+	},
+	process: {
+		parent: allowlistProcessFields,
+		...allowlistProcessFields,
+	},
+	network: {
+		direction: true,
+	},
+	registry: {
+		data: {
+			strings: true,
+		},
+		hive: true,
+		key: true,
+		path: true,
+		value: true,
+	},
+	Target: {
+		process: {
+			parent: allowlistProcessFields,
+			...allowlistProcessFields,
+		},
+	},
+	user: {
+		id: true,
+	},
+};
+
+// Allow list for the data we include in the events. True means that it is deep-cloned
+// blindly. Object contents means that we only copy the fields that appear explicitly in
+// the sub-object.
+export const AllowlistAlertEventFields: AllowlistFields = {
+	'@timestamp': true,
+	agent: true,
+	Endpoint: true,
+	/* eslint-disable @typescript-eslint/naming-convention */
+	Memory_protection: true,
+	Ransomware: true,
+	data_stream: true,
+	ecs: true,
+	elastic: true,
+	// behavioral protection re-nests some field sets under events.*
+	events: allowlistBaseEventFields,
+	rule: {
+		id: true,
+		name: true,
+		ruleset: true,
+		version: true,
+	},
+	host: {
+		os: true,
+	},
+	...allowlistBaseEventFields,
+};
+
+export function copyAllowlistedFields(
```

Comment:
Looks good. Please do some maintenance in 

  - x-pack/plugins/security_solution/server/lib/telemetry/sender.test.ts:227:12)


<details><summary>Test output</summary>
<p>

#### yes, even hidden code blocks!

```js
 FAIL  x-pack/plugins/security_solution/server/lib/telemetry/sender.test.ts
  ● allowlistEventFields › filters top level

    TypeError: (0 , _sender.copyAllowlistedFields) is not a function

      225 |       b1: 'b1',
      226 |     };
    > 227 |     expect(copyAllowlistedFields(allowlist, event)).toStrictEqual({
          |            ^
      228 |       a: 'a',
      229 |       b: 'b',
      230 |     });

      at Object.<anonymous> (x-pack/plugins/security_solution/server/lib/telemetry/sender.test.ts:227:12)

  ● allowlistEventFields › filters nested

    TypeError: (0 , _sender.copyAllowlistedFields) is not a function

      247 |       },
      248 |     };
    > 249 |     expect(copyAllowlistedFields(allowlist, event)).toStrictEqual({
          |            ^
      250 |       a: {
      251 |         a1: 'a1',
      252 |       },

      at Object.<anonymous> (x-pack/plugins/security_solution/server/lib/telemetry/sender.test.ts:249:12)

  ● allowlistEventFields › filters arrays of objects

    TypeError: (0 , _sender.copyAllowlistedFields) is not a function

      288 |       ],
      289 |     };
    > 290 |     expect(copyAllowlistedFields(allowlist, event)).toStrictEqual({
          |            ^
      291 |       a: [
      292 |         {
      293 |           a1: 'a1',

      at Object.<anonymous> (x-pack/plugins/security_solution/server/lib/telemetry/sender.test.ts:290:12)

  ● allowlistEventFields › doesn't create empty objects

    TypeError: (0 , _sender.copyAllowlistedFields) is not a function

      319 |       },
      320 |     };
    > 321 |     expect(copyAllowlistedFields(allowlist, event)).toStrictEqual({
          |            ^
      322 |       a: 'a',
      323 |       b: 'b',
      324 |     });

      at Object.<anonymous> (x-pack/plugins/security_solution/server/lib/telemetry/sender.test.ts:321:12)

 PASS  x-pack/plugins/security_solution/server/lib/telemetry/endpoint_task.test.ts (22.785 s)
 PASS  x-pack/plugins/security_solution/server/lib/telemetry/diagnostic_task.test.ts (22.788 s)

Test Suites: 2 failed, 3 passed, 5 total
Tests:       4 failed, 32 passed, 36 total
Snapshots:   0 total
Time:        24.075 s
Ran all test suites matching /security_solution\/server\/lib\/telemetry/i.
```

</p>
</details>